### ISSUE TITLE
Non-zero exit code with bad command

### ISSUE THREAD

**Issue by ferrouswheel** [npm/npm#4031]
Npm version 1.3.11

Type `npm sperglebergle`

Get the help message, cool. But then returns exit code 0. Should really give an non zero exit code if the `sperglebergle` action doesn't exist.

**Comment by othiym23** [comment]
This landed as 057d9b7:

```
bauchelain% npm glarb || echo "oh shiiii"

Usage: npm <command>

where <command> is one of:
    access, adduser, bin, bugs, c, cache, completion, config,
    ddp, dedupe, deprecate, dist-tag, docs, edit, explore, get,
    help, help-search, i, init, install, install-test, it, link,
    list, ln, logout, ls, outdated, owner, pack, ping, prefix,
    prune, publish, rb, rebuild, repo, restart, root, run,
    run-script, s, se, search, set, shrinkwrap, star, stars,
    start, stop, t, tag, team, test, tst, un, uninstall,
    unpublish, unstar, up, update, v, version, view, whoami

npm <cmd> -h     quick help on <cmd>
npm -l           display full usage info
npm help <term>  search for help on <term>
npm help npm     involved overview

Specify configs in the ini-formatted file:
    /Users/ogd/.npmrc
or on the command line via: npm <command> --key value
Config info can be viewed via: npm help config

npm@3.9.0 /Users/ogd/Documents/projects/npm/npm
oh shiiii
```

Closing as shipped. Thanks for your time!

**Comment by othiym23** [comment]
See also #4780.
